src/Route/RoutePageGenerator.php

```diff
@@ -252,6 +227,45 @@ public function update(SiteInterface $site, OutputInterface $output = null, $cle
         }
     }
 
+    /**
+     * Generate root page (parent for another generated pages).
+     * If root path available in config - create from config
+     * Else - generate it statically.
+     *
+     * @param SiteInterface $site
+     *
+     * @return object root page
+     */
+    public function createRootPage(SiteInterface $site)
+    {
+        foreach ($this->router->getRouteCollection()->all() as $name => $route) {
+            if ('/' === $route->getPath()) {
+                $requirements = $route->getRequirements();
+                $name = trim($name);
+                $root = $this->pageManager->create([
```

Comment:
Why not return here directly? This would make the rest of the code less complex.